app/frontend/components/reportForm/reportForm.tsx

```diff
@@ -0,0 +1,130 @@
+import { Combobox, Transition } from '@headlessui/react'
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+import { ProjectFragment, useProjectsQuery, useWorkHoursQuery } from '../../generated/graphql'
+import { HiCheck, HiSelector } from 'react-icons/hi'
+import { endOfMonth, format, formatISO, parse, startOfMonth } from 'date-fns'
+import { FormattedDuration } from '../duration/formattedDuration'
+
+const ReportForm = () => {
+  const router = useRouter()
+  const [{ data: projectsData }] = useProjectsQuery({ pause: !router.isReady })
+  const [selectedProject, setSelectedProject] = useState<ProjectFragment | undefined>()
+  const [, setProjectQuery] = useState('')
+  const [date, setDate] = useState(format(new Date(), 'yyyy-MM'))
+
+  const parsedDate = parse(date, 'yyyy-MM', new Date())
+
+  const startOfMonthString = formatISO(startOfMonth(parsedDate), { representation: 'date' })
+  const endOfMonthString = formatISO(endOfMonth(parsedDate), { representation: 'date' })
+
+  const [{ data: workHoursData }] = useWorkHoursQuery({
+    variables: { from: startOfMonthString, to: endOfMonthString },
+    pause: !router.isReady,
+  })
+
+  const filteredWorkHour = workHoursData?.workHours.filter((workHour) => workHour.project.id === selectedProject?.id)
+
+  return (
+    <>
+      <div>
+        {
+          <h1 className="mb-4 font-bold">
+            Detailed time report: {startOfMonthString} - {endOfMonthString}
+          </h1>
+        }
+        <h2>Select a project</h2>
+      </div>
+      <div className="flex flex-col">
+        <div className="flex justify-between">
+          <div>
+            <Combobox value={selectedProject} onChange={setSelectedProject}>
+              <div>
+                <div className="relative w-full cursor-default overflow-hidden rounded-lg bg-white text-left shadow-md focus:outline-none focus-visible:ring-2 focus-visible:ring-white focus-visible:ring-opacity-75 focus-visible:ring-offset-2 focus-visible:ring-offset-teal-300 sm:text-sm">
+                  <Combobox.Input<'input', ProjectFragment>
+                    className="w-full border-none py-2 pl-3 pr-10 text-sm leading-5 text-gray-900 focus:ring-0"
+                    displayValue={(project) => project.title}
+                    onChange={(event) => setProjectQuery(event.target.value)}
```

Comment:
Does the combobox work without the `onChange` here?

Reply:
It works, but I think the `OnChange` is required for `Combobox.Input`.